// jumpsuit/lib/render.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Router = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = function (stores, baseComponent, options) {
  var store = (0, _reducer.combine)(_extends({}, stores, {
    routing: _reactRouterRedux.routerReducer
  }), options);

  var syncedHistory = (0, _reactRouterRedux.syncHistoryWithStore)(_reactRouter.browserHistory, store);
  var base = baseComponent.props._isRouteWrapper ? _react2.default.createElement(
    _reactRouter.Router,
    { history: syncedHistory },
    baseComponent
  ) : baseComponent;

  var child = base;
  if (process.env.NODE_ENV !== 'production') {
    if (window.devToolsExtension) {
      console.warn('Jumpsuit doesn\'t support the Redux Dev Tools browser extension!');
    }

    var Hsr = require('./hsr').default;
    var DevTools = require('./devtools').default;

    child = _react2.default.createElement(
      'div',
      null,
      base,
      _react2.default.createElement(DevTools, null),
      _react2.default.createElement(Hsr, null)
    );
  }

  (0, _reactDom.render)(_react2.default.createElement(
    _reactRedux.Provider,
    { store: store },
    child
  ), document.getElementById('app'));
};

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactRouter = require('react-router');

var _reactRedux = require('react-redux');

var _reactRouterRedux = require('react-router-redux');

var _reducer = require('./reducer');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Router = exports.Router = _react2.default.createClass({
  displayName: 'Router',

  propTypes: { children: _react2.default.PropTypes.object },
  getDefaultProps: function getDefaultProps() {
    return { _isRouteWrapper: true };
  },
  render: function render() {
    return _react2.default.createElement(
      'div',
      null,
      undefined.props.children
    );
  }
});
